### ISSUE TITLE
JDK 15 now requires glibc 2.14 impacting support for older OS

### ISSUE THREAD

**Issue by jasontedor** [elastic/elasticsearch#62709]
The tool chain for JDK 15 has been updated to compile again glibc 2.14. This impacts older OS for which this version of glibc is not available via the OS package manager. This means that the bundled JDK will no longer works on these OS which includes Oracle Enterprise Linux 6, RedHat Enterprise Linux 6, and CentOS 6 (the latest is glibc 2.12). We do not want to be blocked from advancing the JDK, as that would mean either regressing to ship with JDK 11, or shipping with an EOL JDK in JDK 14. For such old OS, this is not an acceptable outcome. Our plan is the following:
 - detect when the installed glibc does not meet 2.14
 - if the installed glibc does not meet 2.14, skip using the bundled JDK and fallback to JAVA_HOME
 - mark this as a known issue in our docs, and also in our release blog

Relates elastic/elasticsearch#62635

**Comment by imweijh** [comment]
OpenJDK 15 build by Zulu or Oracle  support CentOS 6

**Comment by rjernst** [comment]
Thanks for the note @imweijh. When originally trying to reproduce this issue, I thought I had verified oracle jdk 15 also failed, but I must have gotten jdks mixed up. I've now verified you are correct, Oracle jdk 15 does continue to work on CentOS 6.

At this time though, I'm not sure we should yet do anything. We switched away from the oracle jdk because of notarization issues on MacOS (https://github.com/elastic/elasticsearch/issues/46498).

I did find a recent issue (https://github.com/AdoptOpenJDK/openjdk-build/issues/2097) in AdoptopenJDK regarding the glibc version. Perhaps they will fix their build chain as suggested there, we will just have to wait and see.
